### PR TITLE
Add check to prevent attaching a profiler when one is already present

### PR DESCRIPTION
**Pull request by davmason**

### CODE REVIEW

**Comment by davmason** [comment]
Fixes #25313

File: src/vm/profilerdiagnosticprotocolhelper.cpp [hunk 1]
```diff
@@ -103,6 +103,14 @@ void ProfilerDiagnosticProtocolHelper::AttachProfiler(DiagnosticsIpc::IpcMessage
         goto ErrExit;
     }
     
+    // The diagnostic server is single threaded, so we do not have to worry about racing
```
Comment (noahfalk) [inline src/vm/profilerdiagnosticprotocolhelper.cpp]: I'd recommend making the ProtocolHelper be responsible only for marshalling serialized buffer -> arguments and return value -> serialized buffer. For any logic related to the operation itself I'd put it inside LoadProfilerForAttach.

Comment (davmason) [reply src/vm/profilerdiagnosticprotocolhelper.cpp]: Yeah, I think that makes more sense

**Comment by davmason** [comment]
@noahfalk I moved the check in to `ProfilingAPIUtility::LoadProfiler`, and it occurred to me that the attach path could race with the startup path since the diagnostics server starts so early. To solve this I used added `ProfilerStatus::kProfStatusPreInitialize` to keep track of if there is already a profiler trying to load. I moved a bunch of logic to a new function so I could aggregate any errors and just set it back to `kProfStatusNone` in one location, but the logic is unchanged.

**Comment by davmason** [comment]
Also meant to mention I did manual verification that attach works and a second attach request is refused, and then ran the profiler tests against it.

**Pull request merged**